### DISCUSSION TITLE
Overriding *_CONFIG configurations

### DISCUSSION THREAD

**Discussion by godwinjoel** [hazelcast/hazelcast#22640] [General]
@ldziedziul, @frant-hartm, @obedsoto

Concerns distribution/src/bin-filemode-755/hz file.

As a user, when I override my config directory using HAZELCAST_CONFIG,
I expected the following -
LOGGING_CONFIG="file:$HAZELCAST_CONFIG/log4j2.properties"

PROMETHEUS="-javaagent:${HAZELCAST_HOME}/lib/jmx_prometheus_javaagent-0.17.2.jar=${PROMETHEUS_PORT}:${HAZELCAST_CONFIG}/jmx_agent_config.yaml"

But it seems all these other configurations still look for relevant files in $HAZELCAST_HOME/config instead of $HAZELCAST_CONFIG
Is this by design? With the current setup, I need to manage different configuration files in at least two other folders.

**Comment by frant-hartm** [comment]
`HAZELCAST_CONFIG` points to the hazelcast.xml/yaml file, not the config directory.
`LOGGING_CONFIG` points to the log4j2 config file.

In your environment, you can setup something like this so you can have it in single directory:

```
HZ_CONFIG_FOLDER=abc...
HAZELCAST_CONFIG=$HZ_CONFIG_FOLDER/config/hazelcast.xml
LOGGING_CONFIG=$HZ_CONFIG_FOLDER/config/log4j2.xml
```

We didn't expect users to be modifying the `jmx_agent_config.yaml` - can you provide more info why you need to do that?

**Reply by godwinjoel** [reply]
I misinterpreted the HAZELCAST_CONFIG as the config directory path, thus expecting all other configurations to use it.

When installing via ansible/puppet, etc we have more control over folders in the system, so idea was to point the config folder to our own directory where updated hazelcast/logging configuration files are available, and other files like the jmx_agent_config are also available in that same location but unchanged.

Thanks for your explanation.
